libnmstate/state.py

```diff
@@ -325,9 +325,16 @@ def verify_interfaces(self, other_state):
 
         self.normalize_for_verification()
         other_state.normalize_for_verification()
+        self._pre_verification_fix(other_state)
 
         self._assert_interfaces_equal(other_state)
 
+    def _pre_verification_fix(self, other_state):
+        for iface_name, iface_state in self.interfaces.items():
+            other_iface_state = other_state.interfaces.get(iface_name)
+            if iface_state.get(Interface.TYPE) == InterfaceType.BOND:
+                bond.iface_state_pre_verify_fix(iface_state, other_iface_state)
```

Comment:
I think this fits in `normalize_for_verification`. We are doing something similar for interface description. When the interface description is empty the NM on-disk config is not saving it so the current state does not contain the description and in `normalize_for_verification` we are removing the empty description from the desire state before verifying.

IMO, this fix seems similar to me so I think `self._pre_verification_fix` should be called there.

Reply:
The `normalize_for_verification()` applies on both desire state and new current state and does not differentiate on that.
For this case, only new current state need to be amended.

Reply:
Fair enough, thanks for explaining.